http-server/src/main/java/org/triplea/server/lobby/game/listing/GameListing.java

```diff
@@ -37,116 +38,50 @@
-    }
-  }
-
-  @VisibleForTesting
-  static class IncorrectApiKey extends RuntimeException {
-    private static final long serialVersionUID = -131279328512375629L;
-
-    IncorrectApiKey() {
-      super("Illegal game modification attempt, access key is incorrect.");
-    }
-  }
-
-  /** Adds a game. Duplicate postings return the same gameId. */
-  String postGame(final String apiKey, final LobbyGame lobbyGame) {
-    final Optional<String> existingGameId =
-        games.entrySet().stream()
-            .filter(entry -> entry.getKey().apiKey.equals(apiKey))
-            .filter(entry -> entry.getValue().equals(lobbyGame))
-            .map(entry -> entry.getKey().id)
-            .findAny();
-    if (existingGameId.isPresent()) {
-      log.warn("Received duplicate game post request for game: {}", existingGameId.get());
-      return existingGameId.get();
-    }
-
+  /** Adds a game. */
+  String postGame(final ApiKey apiKey, final LobbyGame lobbyGame) {
     final String gameId = UUID.randomUUID().toString();
-    gameReaper.registerKeepAlive(gameId);
-    games.put(GameId.builder().id(gameId).apiKey(apiKey).build(), lobbyGame);
-    gameUpdateListener.accept(
-        LobbyGameListing.builder().gameId(gameId).lobbyGame(lobbyGame).build());
+    games.put(new GameId(apiKey, gameId), lobbyGame);
     log.info("Posted game: {}", gameId);
     return gameId;
   }
 
   /** Adds or updates a game. If game is updated, emits a game update event. */
-  void updateGame(final String apiKey, final String gameId, final LobbyGame lobbyGame) {
-    final Map.Entry<GameId, LobbyGame> game =
-        // find game by gameId
-        games.entrySet().stream()
-            .filter(entry -> entry.getKey().id.equals(gameId))
-            .findAny()
-            .orElseThrow(() -> new GameNotFound(gameId));
-    if (!game.getKey().apiKey.equals(apiKey)) {
-      throw new IncorrectApiKey();
-    }
-
-    games.put(game.getKey(), lobbyGame);
-    gameUpdateListener.accept(
-        LobbyGameListing.builder().gameId(game.getKey().id).lobbyGame(lobbyGame).build());
-  }
-
-  void removeGame(final String apiKey, final String gameId) {
-    final var game =
-        games.entrySet().stream().filter(entry -> entry.getKey().id.equals(gameId)).findAny();
-    game.ifPresent(
-        gameEntryToRemove -> {
-          final GameId gameIdToRemove = gameEntryToRemove.getKey();
-          if (!gameIdToRemove.apiKey.equals(apiKey)) {
-            throw new IncorrectApiKey();
-          }
-          removeGame(gameIdToRemove);
-        });
+  boolean updateGame(final ApiKey apiKey, final String gameId, final LobbyGame lobbyGame) {
+    final GameId listedGameId = new GameId(apiKey, gameId);
+    return Optional.ofNullable(games.getIfPresent(listedGameId))
+        .map(
+            game -> {
+              games.put(listedGameId, lobbyGame);
+              return true;
+            })
+        .orElse(false);
```

Comment:
Haven't actually tried this, but what about this:
```java
return games.asMap().replace(listedGameId, lobbyGame) != null
```

Reply:
Good suggestion, I'd break it up into two lines though, we're doing many different operations on the single line, bit confusing.

Reply:
Updated (3fb6320f1) to this:
```
    final var listedGameId = new GameId(apiKey, gameId);
    final var existingValue = games.asMap().replace(listedGameId, lobbyGame);
    return existingValue != null;
```